tests/test_core_commands/test_gs_commit.py

```diff
@@ -0,0 +1,98 @@
+import os
+import subprocess
+
+from GitSavvy.core import git_command
+from GitSavvy.tests.common import GitRepoTestCase
+from GitSavvy.common.util import super_key
+
+
+class CommonCommitMixin(object):
+    """ Help methods for commit command testing """
+
+    def get_view(self):
+        """
+        Get the currently active view, set it to self.view and
+        return it's line_count
+        """
+        self.view = self.window.active_view()
+        line_count = self.view.rowcol(self.view.size())[0]
+        # print('Got view: %s, line count: %s' % (self.view, line_count))
+        return line_count
+
+    def assert_commit_view_rendered(self, first_line=None):
+        """ Assert the commit view is correctly rendered """
+        yield self.get_view  # yield back control when view has any lines
+        first_rows = self.get_rows(*range(0, 2))
+        self.assertEqual(first_rows, [first_line or '', ''])
+
+        first_comment_row = self.get_row(2)
+        self.assertEqual(first_comment_row, '## To make a commit, type your '
+                         'commit message and press {key}-ENTER. To cancel'
+                         .format(key=super_key))
+
+    def finish_commit(self, message=None):
+        """
+        Execute the commit while inserting the
+        given message in the start of the buffer
+        """
+        if message:
+            self.view.run_command("insert", {"characters": message}),
+        self.window.run_command('gs_commit_view_do_commit')
+        yield lambda: not self.window.views()  # yield control when view closed
+
+    def assert_commit_log_updated(self, message, count=2):
+        """
+        Assert commit was performed correctly by inspecting last commit
+        message in log as well as the commit count
+        """
+        output = subprocess.check_output(["git", "log", "--oneline"],
+                                         cwd=self._temp_dir)
+        commits = output.splitlines()
+        self.assertEqual(len(commits), 2)
+        self.assertIn(message, commits[0].decode())
+
+
+class TestCommit(CommonCommitMixin, GitRepoTestCase, git_command.GitCommand):
+
+    def test_add_and_commit_file(self):
+        """ Test adding a new file and committing """
+        foo = os.path.join(self.repo_path, "foo")
+        with open(foo, "w") as f:
+            f.write("foo")
+
+        self.stage_file(foo)
+        self.window.run_command("gs_commit")
+
+        yield from self.assert_commit_view_rendered(first_line=None)
```

Comment:
This specific `yield from` calls into every yield command in the `self.assert_commit_view_rendered` method. Without this `yield from`, simply calling the method would not call the yield expression inside the method, thus failing the test.

For more information see here [the changelog description in Python 3.3][1].

The reason we use yield at all here, is due to the commands' use of `sublime.set_timeout_async` which basically defers and returns control back to the text thread. This a feature of @randy3k 's [excellent UnitTesting plugin][2].

[1]: https://docs.python.org/3/whatsnew/3.3.html#pep-380-syntax-for-delegating-to-a-subgenerator
[2]: https://github.com/randy3k/UnitTesting-example/tree/090f271592788d584d41b92e810b0c7577e066e9